### PR TITLE
CMake: bring mungefs closer to Normal and Boring

### PR DESCRIPTION
**Pull request by SwooshyCueb**
Addresses #31, plus some CMake TLC

### CODE REVIEW

File: cmake/FindAvroCpp.cmake [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  AvroCpp library to be linked
+``AVROCPP_AVROGEN_EXECUTABLE``
+  path to avrogencpp tool
+
+TODO
+^^^^
+
+* Components
+  * library
+  * executable
```
Comment (korydraughn) [inline cmake/FindAvroCpp.cmake:36]: Are you planning to resolve the TODO in this PR too?

File: cmake/FindZeroMQ.cmake [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  true if ZeroMQ headers and library were found
+``ZeroMQ_INCLUDE_DIR``
+  the directory containing ZeroMQ headers
+``ZeroMQ_LIBRARY``
+  ZeroMQ library to be linked
+
+TODO
+^^^^
+
+* Version matching
```
Comment (korydraughn) [inline cmake/FindZeroMQ.cmake:30]: Same here.

File: cmake/Findcppzmq.cmake [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+``cppzmq_FOUND``
+  true if cppzmq headers were found
+``cppzmq_INCLUDE_DIR``
+  the directory containing cppzmq headers
+
+TODO
+^^^^
+
+* Version matching
```
Comment (korydraughn) [inline cmake/Findcppzmq.cmake:28]: Same here.

File: CMakeLists.txt [hunk 4]
```diff
@@ -210,72 +100,31 @@ set (
...
-file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/include)
+file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/include")
 
 add_custom_command(
-   OUTPUT ${CMAKE_BINARY_DIR}/include/${AVRO_FILE}.hpp
-   COMMAND LD_LIBRARY_PATH=${BOOST_PATH}/lib:${CLANG_LIBS} ${IRODS_EXTERNALS_FULLPATH_AVRO}/bin/avrogencpp -o ${CMAKE_BINARY_DIR}/include/${AVRO_FILE}.hpp -i ${CMAKE_SOURCE_DIR}/avro_schemas/${AVRO_FILE}.json
-   MAIN_DEPENDENCY ${CMAKE_SOURCE_DIR}/avro_schemas/${AVRO_FILE}.json
+   OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/include/${AVRO_FILE}.hpp"
+   COMMAND "${AVROCPP_AVROGEN_EXECUTABLE}" -o "${CMAKE_CURRENT_BINARY_DIR}/include/${AVRO_FILE}.hpp" -i "${CMAKE_CURRENT_SOURCE_DIR}/avro_schemas/${AVRO_FILE}.json"
+   MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/avro_schemas/${AVRO_FILE}.json"
```
Comment (korydraughn) [inline CMakeLists.txt:108]: So the find module for avro and the new avro target is why we no longer have to use `LD_LIBRARY_PATH`, correct?

**Review by korydraughn — COMMENTED** [review]
Looks fine on my first pass.

Will do a second pass at a later point.

File: CMakeLists.txt (hunk 4 above)
Comment (SwooshyCueb) [reply CMakeLists.txt:108]: Yes and no.

Externals will still need `LD_LIBRARY_PATH` for as long as our `avrogencpp` doesn't have a proper `RPATH`/`RUNPATH`, but it should be set before calling CMake, not *in* CMakeLists.txt.

File: cmake/FindAvroCpp.cmake (hunk 1 above)
Comment (SwooshyCueb) [reply cmake/FindAvroCpp.cmake:36]: I'll probably do this when I move the find module into the irods repository.

File: cmake/FindZeroMQ.cmake (hunk 2 above)
Comment (SwooshyCueb) [reply cmake/FindZeroMQ.cmake:30]: I'll probably do this when I move the find module into the irods repository.

File: cmake/Findcppzmq.cmake (hunk 3 above)
Comment (SwooshyCueb) [reply cmake/Findcppzmq.cmake:28]: I'll probably do this when I move the find module into the irods repository.

File: CMakeLists.txt (hunk 4 above)
Comment (korydraughn) [reply CMakeLists.txt:108]: Makes sense.

I'm assuming externals will need a tweak to address the LD_LIBRARY_PATH-before-cmake then?

Comment (SwooshyCueb) [reply CMakeLists.txt:108]: Yes, among other things

Comment (korydraughn) [reply CMakeLists.txt:108]: Cool cool.

**Comment by SwooshyCueb** [comment]
Marking this as ready for review. I considered changing up how the generated header file to be its own CMake target like we do with the control plane, but decided against it.

**Comment by SwooshyCueb** [comment]
Also, if there are no objections, I'm going to bump the version, so that once this is merged we have a commit we can tag that we can point externals at.

**Comment by alanking** [comment]
> Also, if there are no objections, I'm going to bump the version, so that once this is merged we have a commit we can tag that we can point externals at.

Seems fine to me.

**Comment by korydraughn** [comment]
Bumping the version sounds fine.

**Review by alanking — COMMENTED** [review]
Nothing jumps out at me in the find modules. I say bump version and squash to taste (omit #s)

**Comment by korydraughn** [comment]
Agreed.

**Comment by korydraughn** [comment]
Let's get some issue numbers attached to all commits or squash things down further.

**Comment by SwooshyCueb** [comment]
Added numbers. Not sure if it makes sense to have them on the version bump commit though, nor does it really make sense to me to squash the version bump commit into another commit

**Comment by korydraughn** [comment]
I agree. Create an issue for the version bump. You can include the motivation behind the bump in the issue and commit. Either is fine to me.

**Comment by SwooshyCueb** [comment]
Created an issue for the version bump. The commit points to it now.

**Comment by korydraughn** [comment]
Add the pounds.

**Review by korydraughn — APPROVED** [review]

**Review by alanking — APPROVED** [review]
Very nice. Let's # it up

**Comment by SwooshyCueb** [comment]
\#'d

**Comment by SwooshyCueb** [comment]
Remember to tag after merging

**Comment by korydraughn** [comment]
On it.

**Comment by korydraughn** [comment]
Done.

**Pull request merged**